Index: shinken/shinken/modules/collectd_arbiter.py
===================================================================
--- shinken.orig/shinken/modules/collectd_arbiter.py
+++ shinken/shinken/modules/collectd_arbiter.py
@@ -69,7 +69,6 @@ def get_instance(plugin):
         port = DEFAULT_PORT
 
     if hasattr(plugin, 'grouped_collectd_plugins'):
-        grouped_collectd_plugins = plugin.grouped_collectd_plugins
         grouped_collectd_plugins = [name.strip()
                                     for name in plugin.grouped_collectd_plugins.split(',')]
     else:
