diff --git a/shinken/modules/collectd_arbiter.py b/shinken/modules/collectd_arbiter.py
index f3faca6..441c55e 100644
--- a/shinken/modules/collectd_arbiter.py
+++ b/shinken/modules/collectd_arbiter.py
@@ -68,7 +68,6 @@ def get_instance(plugin):
         port = DEFAULT_PORT
 
     if hasattr(plugin, 'grouped_collectd_plugins'):
-        grouped_collectd_plugins = plugin.grouped_collectd_plugins
         grouped_collectd_plugins = [name.strip()
                                     for name in plugin.grouped_collectd_plugins.split(',')]
     else:
